lib/pbench/server/api/resources/query_apis/query_dataset.py

```diff
@@ -0,0 +1,268 @@
+    names that will contain the data. (Note that the dashboard uses the
+    same date range it used to find controllers; however it could make this
+    more efficient by using only the year and month in which the result
+    was generated.)
+
+    Note that although the name "queryResult" is retained for consistency
+    with the existing dashboard UI code, the word "result" is a bit
+    misleading in that this deals with Pbench "run" documents, not Pbench
+    "results" documents. It's a follow-on to queryResults, which returns
+    the list of datasets within a specified date range for a given controller
+    (as reported by queryControllers). The queryResult API returns details from
+    the dataset run document; basically the union of the "run" and "@metadata"
+    sub-documents.
+    """
+
+    def __init__(self, config, logger):
+        self.logger = logger
+        self.es_url = get_es_url(config)
+        self.prefix = get_index_prefix(config)
+
+    def post(self):
+        """
+        POST to get details for a specific Pbench dataset within the
+        specified date range, and owned by a specified user:
+
+        {
+            "user": "email-tag",
+            "name": "dataset-name",
+            "start": "start-time",
+            "end": "end-time"
+        }
+
+        "user" specifies the owner of the data to be searched; it need not
+        necessarily be the user represented by the session token header,
+        assuming the session user is authorized to view "user"s data. This
+        will be used to constrain the Elasticsearch query.
+
+        TODO: When we have authorization infrastructure, we'll need to
+        check that "session user" has rights to view "user" data. We might
+        also default a missing "user" JSON field with the authorization
+        token's user. (Or with the default user "public" if there's no
+        token, indicating there's no logged-in user.)
+
+        "name" is the name of a Pbench agent dataset (tarball).
+
+        "start" is a time string representing the starting range of dates
+        to search.
+
+        "end" is a time string representing the end range of dates to
+        to search.
+
+        The Elasticsearch URL string is also dependent upon the configured
+        "prefix". This is a Pbench artifact that allows multiple Pbench
+        instances to share a single Elasticsearch server, by segregating
+        their data into unique namespaces. Each Pbench index name is
+        qualified by the index_prefix of the Pbench server that originally
+        indexed the data. The prefix comes from the pbench-server.cfg
+        file for the server instance. For example, production, "staging",
+        and development server instances might all share one Elasticsearch
+        cluster by choosing unique prefixes to generate unique index
+        names.
+
+        Required headers include
+
+        X-auth-token:   Pbench session authorization token authorized to access
```

Comment:
This should be bearer `Authorization`

Reply:
Yep, another stale comment. I want to refactor all this to reduce duplication; I've been struggling to put that off until I'm done here.